containers/python/rpc.py

```diff
@@ -499,39 +507,88 @@ def start(self, model, host, port, model_name, model_version, input_type):
+
+            if not metric_type and not metric_description:
+                raise Exception(
+                    "{}: Metric Type and Metric Description are Required in Config File.".
+                    format(name))
+
+            if metric_type == 'Counter':
+                self.metrics[name] = Counter(prefix + name, metric_description)
+            elif metric_type == 'Gauge':
+                self.metrics[name] = Gauge(prefix + name, metric_description)
+            elif metric_type == 'Histogram':
+                if 'bucket' in spec.keys():
+                    buckets = np.linspace(
+                        *spec['bucket']).tolist() + [float("inf")]
+                    self.metrics[name] = Histogram(
+                        prefix + name, metric_description, buckets=buckets)
+                else:
+                    self.metrics[name] = Histogram(prefix + name,
+                                                   metric_description)
+            elif metric_type == 'Summary':
+                self.metrics[name] = Summary(prefix + name, metric_description)
+            else:
+                raise Exception(
+                    "Unknown Metric Type: {}. See config file.".format(
+                        metric_type))
+
+            self.name_to_type[name] = metric_type
 
     def collect(self):
-        latest_metric_dict = None
-        while self.pipe_conn.poll():
-            latest_metric_dict = self.pipe_conn.recv()
-        if latest_metric_dict:
-            for name, val in latest_metric_dict.items():
-                try:
-                    yield GaugeMetricFamily(
-                        name=name,
-                        documentation=name,  # Required Argument
-                        value=val)
-                except ValueError:
-                    pass
+        while True:
+            latest_metric_dict = self.pipe_conn.recv(
+            )  # This call is blocking.
+            for name, value in latest_metric_dict.items():
+                metric = self.metrics[name]
+                if self.name_to_type[name] == 'Counter':
+                    metric.inc(value)
+                elif self.name_to_type[name] == 'Gauge':
+                    metric.set(value)
+                elif self.name_to_type[name] == 'Histogram' or self.name_to_type[name] == 'Summary':
+                    metric.observe(value)
+                else:
+                    raise Exception(
+                        "Unknown Metric Type for {}. See config file.".format(
+                            name))
 
 
 def run_metric(child_conn):
     """
     This function takes a child_conn at the end of the pipe and
     receive object to update prometheus metric.
 
-    It is recommended to be ran in a separate process. 
+    It is recommended to be ran in a separate process.
     """
-    REGISTRY.register(MetricCollector(child_conn))
+    collector = MetricCollector(child_conn)
     start_http_server(1390)
```

Comment:
Nope. This port does not need to be exposed to the user. Prometheus can detect and access this port since the Prometheus container and model container are technically in the same "pod". This might need to change to k8s though.